### PR TITLE
Add AutoScalerRef in TidbCluster Status

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

Close https://github.com/pingcap/tidb-operator/issues/2484

Add Auto-Scaler reference in TidbCluster Status which can avoid avoiding 2 auto-scaler directly take effect to the same tidbcluster. 

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Support Auto-Scaler Reference in `TidbCluster` Status when there existed `TidbClusterAutoScaler`
```

### CODE REVIEW

File: pkg/label/label.go [hunk 1]
```diff
@@ -117,18 +117,6 @@ const (
...
-	// AnnTiDBConsecutiveScaleInCount describes the least consecutive count to scale-in for tidb
-	AnnTiDBConsecutiveScaleInCount = "tidb.tidb.pingcap.com/consecutive-scale-in-count"
-	// AnnTiKVConsecutiveScaleOutCount describes the least consecutive count to scale-out for tikv
-	AnnTiKVConsecutiveScaleOutCount = "tikv.tidb.pingcap.com/consecutive-scale-out-count"
-	// AnnTiKVConsecutiveScaleInCount describes the least consecutive count to scale-in for tikv
-	AnnTiKVConsecutiveScaleInCount = "tikv.tidb.pingcap.com/consecutive-scale-in-count"
-	// AnnAutoScalingTargetName describes the target TidbCluster Ref Name for the TidbCluserAutoScaler
-	AnnAutoScalingTargetName = "auto-scaling.tidb.pingcap.com/target-name"
-	// AnnAutoScalingTargetNamespace describes the target TidbCluster Ref Namespace for the TidbCluserAutoScaler
-	AnnAutoScalingTargetNamespace = "auto-scaling.tidb.pingcap.com/target-namespace"
```
Comment (Yisaer) [inline pkg/label/label.go:131]: remove useless code

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 2]
```diff
@@ -135,21 +135,6 @@ type BasicAutoScalerSpec struct {
...
-	// +optional
-	ScaleOutThreshold *int32 `json:"scaleOutThreshold,omitempty"`
-
-	// ScaleInThreshold describe the consecutive threshold for the auto-scaling,
-	// if the consecutive counts of the scale-in result in auto-scaling reach this number,
-	// the auto-scaling would be performed.
-	// If not set, the default value is 5.
-	// +optional
-	ScaleInThreshold *int32 `json:"scaleInThreshold,omitempty"`
-
```
Comment (Yisaer) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:152]: remove useless code

File: pkg/manager/member/tidbcluster_status_manager.go [hunk 3]
```diff
@@ -120,6 +124,23 @@ func (tcsm *TidbClusterStatusManager) syncDashboardMetricStorage(tc *v1alpha1.Ti
 	return nil
 }
 
+func (tcsm *TidbClusterStatusManager) syncAutoScalerRef(tc *v1alpha1.TidbCluster) error {
+	if tc.Status.AutoScaler == nil {
+		return nil
+	}
+	tacNamespace := tc.Status.AutoScaler.Namespace
+	tacName := tc.Status.AutoScaler.Name
+	_, err := tcsm.cli.PingcapV1alpha1().TidbClusterAutoScalers(tacNamespace).Get(tacName, metav1.GetOptions{})
```
Comment (zjj2wry) [inline pkg/manager/member/tidbcluster_status_manager.go:133]: why not use lister?

Comment (DanielZhangQD) [reply pkg/manager/member/tidbcluster_status_manager.go:133]: +1, ever raised the same comment for TidbMonitor [here](https://github.com/pingcap/tidb-operator/pull/2424#discussion_r422982794)

Comment (Yisaer) [reply pkg/manager/member/tidbcluster_status_manager.go:133]: good catch.

Comment (Yisaer) [reply pkg/manager/member/tidbcluster_status_manager.go]: updated, and also add the unit test

File: pkg/autoscaler/autoscaler/util.go [hunk 4]
```diff
@@ -59,36 +50,6 @@ func checkStsAutoScalingPrerequisites(set *appsv1.StatefulSet) bool {
 	return true
 }
 
-func checkStsAutoScaling(tac *v1alpha1.TidbClusterAutoScaler, thresholdSeconds, intervalSeconds int32, memberType v1alpha1.MemberType) (bool, error) {
-	realClock := clockwork.NewRealClock()
-	if tac.Annotations == nil {
-		tac.Annotations = map[string]string{}
-	}
```
Comment (Yisaer) [inline pkg/autoscaler/autoscaler/util.go:66]: This function only used for tikv, move to the `checkTiKVStsAutoScaling`

File: pkg/autoscaler/autoscaler/util.go [hunk 5]
```diff
@@ -251,29 +215,6 @@ func defaultTAC(tac *v1alpha1.TidbClusterAutoScaler) {
...
-	name := tac.Annotations[label.AnnAutoScalingTargetName]
-	namespace := tac.Annotations[label.AnnAutoScalingTargetNamespace]
-	if name == tac.Spec.Cluster.Name && namespace == tac.Spec.Cluster.Namespace {
-		return
-	}
-	// If not satisfied, reset tac Ann
-	resetAutoScalingAnn(tac)
-}
-
```
Comment (Yisaer) [inline pkg/autoscaler/autoscaler/util.go:276]: use patch reference instead of recording in annotations.

**Pull request merged**